common/utils/start.sh

```diff
@@ -23,90 +21,237 @@ function check_reverse_proxy {
     fi
 }
 
-# For JAR based applications (APP_MAIN_CLASS set), look for a single JAR file if APP_FILE
-# is not set and use that. If there is not exactly 1 jar APP_FILE will remain unset.
-# For Python applications, look for a single .py file
-if [ -z "$APP_FILE" ]; then
-    if [ -n "$APP_MAIN_CLASS" ]; then
-        cnt=$(cd $APP_ROOT/src/; ls -1 *.jar | wc -l)
-        if [ "$cnt" -eq "1" ]; then
-            APP_FILE=$(cd $APP_ROOT/src/; ls *.jar)
+function get_deployment {
+    # If this fails because the file is not there or the label isn't present, then
+    # DEPLOYMENT will be an empty string
+    DEPLOYMENT=$(grep "^deployment=" /etc/podinfo/labels | cut -d'=' -f2- | tr -d '"')
+}
+
+function delete_ephemeral {
+    local appstatus=$1
+    local line
+    echo "Deleting cluster $OSHINKO_CLUSTER_NAME"
+    line=$($CLI delete $OSHINKO_CLUSTER_NAME --app=$POD_NAME --app-status=$1 $CLI_ARGS 2>&1)
+    echo $line
+}
+
+function handle_term {
+    echo Received a termination signal
+
+    # If we've saved a PID for a subprocess, kill that first before
+    # trying to delete the cluster
+    local cnt
+    local killed=1
+    if [ -n "$PID" ]; then
```

Comment:
These changes are here because during testing I saw spark-submit ignore a termination signal.  So, now we send TERM and then wait for 10 seconds and follow up with a KILL if it's still running.

kill -0 $PID effectively translates to "is the process still alive"